### PR TITLE
mgr/cephadm: adding spec fields for oauth2-proxy whitelist_domains

### PR DESCRIPTION
**Pull request by rkachach**
this field is needed in order to configure which domains are allowed for redirection during login and/or logout

Fixes: https://tracker.ceph.com/issues/67934

Configuration example:

```
service_type: oauth2-proxy
placement:
  hosts:
    - ceph-node-2
spec:
  https_address: "0.0.0.0:4180"
  provider_display_name: "My OIDC Provider"
  client_id: "oauth2-proxy"
  client_secret: "vwCLTITb2rtem583523uXcuSNItDsz1I"
  cookie_secret: "kbAEM9opAmuHskQvt0AW8oeJRaOM2BYy5Loba0kZ0SQ="
  oidc_issuer_url: "http://192.168.100.1:8080/realms/ceph"
  whitelist_domains:
    - domain1.com
    - 192.168.100.1:8080

```

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [x] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [ ] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by rkachach** [comment]
jenkins test make check

**Comment by rkachach** [comment]
jenkins test make check

**Review by adk3798 — APPROVED** [review]

**Comment by adk3798** [comment]
https://pulpito.ceph.com/adking-2024-09-10_02:59:23-orch:cephadm-wip-adk-testing-2024-09-09-1904-distro-default-smithi/

Failures:
- stray daemon warning in smb test, known issue
- `Test failure: test_nfs_export_creation_without_cmount_path (tasks.cephfs.test_nfs.TestNFS)` caused by a PR in the run
- 2x agent on staggered upgrade failure, known issue
- deploy raw test timed out, known issue
- `test_rgw_multisite` test failed to generate token on realm bootstrap, somewhat new but not the first time I've seen it and it's unrelated to PRs in this run
- 4x mds_upgrade_sequence failures, known issue
- rgw-ingress tests timed out, known issue

Nothing to block merging PRs in the run other than the NFS failure related one

File: src/pybind/mgr/cephadm/services/oauth2_proxy.py [hunk 1]
```diff
@@ -67,10 +67,12 @@ def generate_random_secret(self) -> str:
     def generate_config(self, daemon_spec: CephadmDaemonDeploySpec) -> Tuple[Dict[str, Any], List[str]]:
         assert self.TYPE == daemon_spec.daemon_type
         svc_spec = cast(OAuth2ProxySpec, self.mgr.spec_store[daemon_spec.service_name].spec)
+        whitelist_domains = svc_spec.whitelist_domains or []
+        whitelist_domains += self.get_service_ips_and_hosts('mgmt-gateway')
         context = {
             'spec': svc_spec,
             'cookie_secret': svc_spec.cookie_secret or self.generate_random_secret(),
-            'whitelist_domains': self.get_service_ips_and_hosts('mgmt-gateway'),
+            'whitelist_domains': whitelist_domains,
```
[the remark this answers is not in the document]
Comment (rkachach) [reply src/pybind/mgr/cephadm/services/oauth2_proxy.py:75]: Thanks @epuertat for pointing that out. The thing is that in this case the spec is replicating the [configuration of the oauth2-proxy](https://oauth2-proxy.github.io/oauth2-proxy/configuration/overview/) which is using `whitelist_domains`... do you still think we should change in (even we later generate whitelist for oauth2-proxy real configuration) ?

[the remark this answers is not in the document]
Comment (rkachach) [reply src/pybind/mgr/cephadm/services/oauth2_proxy.py:75]: @adk3798 I had discussion on slack with @epuertat and we agreed to leave the field as it is and talk to oauth2-proxy project to see if they can change the field name on their side.

[the remark this answers is not in the document]
Comment (epuertat) [reply src/pybind/mgr/cephadm/services/oauth2_proxy.py:75]: @rkachach sorry for the misunderstanding from my side: I meant that I was ok with asking the oauth2 dev to change that, but from the Ceph side we shouldn't use that setting name. That's my view according to Ceph standards.

[the remark this answers is not in the document]
Comment (rkachach) [reply src/pybind/mgr/cephadm/services/oauth2_proxy.py:75]: @epuertat my bad! I understand the contrary :see_no_evil:  .. I'll open a new PR to fix it.

**Pull request merged**